### DISCUSSION TITLE
Detrended Price Oscillator (DPO)

### DISCUSSION THREAD

**Discussion by DaveSkender** [DaveSkender/Stock.Indicators#551] [Indicators]
Community discussion about [Detrended Price Oscillator (DPO)](https://dotnet.stockindicators.dev/indicators/Dpo/#content)

**Comment by ksathur** [comment]
When I calculate DPO value for the data in a csv file, I think it uses the future data to computer the DPO value at the current price point.

**Reply by DaveSkender** [reply]
This one is a bit unusual as it shifts the values back by `N/2+1` periods, so every DPO value is definitively using future prices, by design.  This is also why you’ll never see the last few bars calculated.
